### PR TITLE
feat: print info if some of the filters don't match any projects

### LINKED ISSUE #2320: Print an info message if some of the filters don't match any projects
If one of the filters don't match any projects in the workspace, print an info message about it.

In some, cases, a warning may be printed. For instance, when filtering by directory or package name.

### PR DESCRIPTION
**Pull request by yaodingyd**
Close #2320

### CODE REVIEW

**Comment by yaodingyd** [comment]
What's the correct way to do info/warning?

**Comment by zkochan** [comment]
The easiest way to add logging would be to install `@pnpm/logger` as a peer dep and user `logger.info()`. But I think that would make this package a bit less generic.

I think the next solution would be nicer. Instead of logging directly in the `@pnpm/filter-workspace-packages` package, just add all the unmatched selectors into a list and return both the selections and the array of unmatched filters. I mean here: https://github.com/pnpm/pnpm/blob/10020d5cff32300347fcfadb494f10290e80828b/packages/filter-workspace-packages/src/index.ts#L112

This way, you'll be able to log the unmatched filters in the main pnpm package, where this filtering is used.

File: packages/pnpm/src/main.ts [hunk 1]
```diff
@@ -124,17 +124,24 @@ export default async function run (inputArgv: string[]) {
...
       prefix: process.cwd(),
       workspaceDir: wsDir,
     })
+    config.selectedProjectsGraph = filterResults.selectedProjectsGraph
     if (R.isEmpty(config.selectedProjectsGraph)) {
       if (!config['parseable']) {
         console.log(`No projects matched the filters in "${wsDir}"`)
       }
       process.exit(0)
       return
```
Comment (yaodingyd) [inline packages/pnpm/src/main.ts:137]: Do we need a return after the `process.exit(0)`? is it reachable code?

Comment (zkochan) [reply packages/pnpm/src/main.ts:137]: not needed but the linter is silly, so it will report a warning if you remove it

**Comment by zkochan** [comment]
Looks good but some tests are failing

**Pull request merged**